site/docs/components/_index/ComponentsList.tsx

```diff
@@ -63,28 +88,99 @@ const ComponentStatusData = ({
   );
 };
 
+const ComponentHeader = ({
+  logo,
+  label,
+  isSorted,
+  ascendingOrder,
+}: {
+  logo: JSX.Element;
+  label: string;
+  isSorted: boolean;
+  ascendingOrder: boolean;
+}) => {
+  const arrowIcon = ascendingOrder ? <ArrowUpIcon /> : <ArrowDownIcon />;
+  return (
+    <Button>
+      <div className={styles.headerContainer}>
+        <div>
+          {logo}
+          <span>{label}</span>
+        </div>
+        {isSorted && arrowIcon}
+      </div>
+    </Button>
+  );
+};
+
 const ComponentsList = () => {
+  const [componentsList, setComponentsList] = useState(
+    componentsListSorting("devStatus", true)
+  );
+
+  const [isSortedBy, setIsSortedBy] = useState<SortedBy>("devStatus");
+
+  const [hasAscendingOrder, setHasAscendingOrder] = useState(true);
+
+  const handleSorting = (componentDetail: SortedBy) => {
+    setIsSortedBy(componentDetail);
+
+    if (componentDetail === isSortedBy) {
+      setHasAscendingOrder((hasAscendingOrder) => !hasAscendingOrder);
+      setComponentsList(
+        componentsListSorting(componentDetail, !hasAscendingOrder)
+      );
+    } else {
+      // we want to reset the sorting order when switching to a different column
+      setHasAscendingOrder(true);
+      setComponentsList(componentsListSorting(componentDetail, true));
+    }
+  };
+
+  const ariaSort = hasAscendingOrder ? "ascending" : "descending";
+
   return (
     <div className={styles.componentList}>
       <table>
         <thead>
           <tr>
-            <th>
-              <StorybookLogo />
-              <span>Component</span>
+            <th
+              onClick={() => handleSorting("name")}
```

Comment:
Wouldn't it be neater to put the `onClick` be on the `<Button>` in `<ComponentHeader>`? That's the actual interactive and focusable element that's being clicked (or potentially activated via keyboard), not the `<th>`.

How you've done it still seems to work (even via keybaord). I suspect that's due to the button's click event bubbling up to the table header. But, I think it make the code easier to understand and maintain if we move the `onClick` to the component that's actually being clicked.

Reply:
Makes sense, I've added the Button after our chat on Friday so forgot to move the `onClick`